docs/sources/developers/kinds/maturity.md

```diff
@@ -0,0 +1,403 @@
+
+This one is tricky to think about, and requires some background.
+
+Historically, Grafana’s dashboard JSON has often contained fields with the explicit value `null`. 
+This was problematic, because explicit `null` introduces an ambiguity: is a JSON field being present 
+with value null meaningfully different from the field being absent? That is, should a program behave differently 
+if it encounters a null vs. an absent field?
+
+In almost all cases, the answer is “no.” Thus, the ambiguity: if both explicit null and absence are *accepted* 
+by a system, it pushes responsibility onto anyone writing code in that system to decide, case-by-case, 
+whether the two are *intended to be meaningfully different*, and therefore whether behavior should be different.
+
+CUE does have a `null` type, and only accepts data containing `nulls` as valid if the schema explicitly allows a `null`.
+That means, by default, using CUE for schemas removes the possibility of ambiguity in code that receives data validated 
+by those schemas, even if the language they’re writing in still allows for ambiguity. (Javascript does, Go doesn’t.)
+
+As a schema author, this means you’re being unambiguous by default - no `nulls`. That’s good! The only question is 
+whether it’s worth explicitly allowing a `null` for some particular case:
+~~~ Cue
+someField: int32 | null
+~~~
+
+The *only* time this *may* be a good idea is if your field needs to be able to represent a value 
+that is not otherwise acceptable within the value space - for example, if `someField` needs to be able to contain 
+[Infinity](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/POSITIVE_INFINITY). 
+When such values are serialized to null by default, it can be convenient to accept null in the schema - but even then, 
+explicit null is unlikely to be the best way to represent such values, because it is so subtle and falsey.
+
+**Above all, DO NOT accept `null` in a schema simply because current behavior sometimes unintentionally produces a `null`.** 
+Schematization is an opportunity to get rid of this ambiguity. Fix the accidental null-producing behavior, instead.
+
+## Setting Defaults and Field Optionality
+
+**TODO oof is this a complicated topic**
+
+## Comments and Documentation
+
+Kind declarations are used to generate a variety of documentation. This is a superpower of schema-centric 
+development - certain key documentation can automatically be kept in line with the code. 
+
+But, that documentation can only be as good as the source from which it’s generated. This section offers guidance 
+on how to write effective documentation for kinds in the kind declarations themselves.
+
+### Kind-level documentation
+TODO providing a high-level summary of the type
+TODO link to examples of where it shows up
+
+### Field-level documentation
+TODO criteria for writing good field-level docs
+TODO link to examples of where it shows up**
```

Comment:
```suggestion
```

Reply:
> **TODO oof is this a complicated topic**
:joy:

Reply:
yeah, and i have to write a quick treatment of it today, in another doc :) (one we'll eventually move out to be public-facing, like these)